src/plane.rs

```diff
@@ -316,28 +316,27 @@ impl<'a, T: Pixel> ExactSizeIterator for IterWidth<'a, T> { }
 impl<'a, T: Pixel> FusedIterator for IterWidth<'a, T> { }
 
 pub struct RowsIter<'a, T: Pixel> {
-  ps: &'a PlaneSlice<'a, T>,
-  next_row: usize,
+  plane: &'a Plane<T>,
+  x: isize,
+  y: isize,
 }
 
 impl<'a, T: Pixel> Iterator for RowsIter<'a, T> {
   type Item = &'a [T];
 
   fn next(&mut self) -> Option<Self::Item> {
-    let remaining = self.ps.plane.cfg.height as isize - self.ps.y + self.next_row as isize;
-    if remaining > 0 {
-      let row = self.next_row;
-      self.next_row += 1;
+    if self.plane.cfg.height as isize > self.y {
       // cannot directly return self.ps.row(row) due to lifetime issue
-      let range = self.ps.slice_range(row);
-      Some(&self.ps.plane.data[range])
+      let range = PlaneSlice::slice_range(&self.plane, self.x, self.y);
```

Comment:
Would it make more sense for slice_range to be a method on plane?

Reply:
Yes, you're right. Now that it is refactored that way, it is not related to `PlaneSlice` anymore.